src/main/java/de/maxhenkel/reap/TreeEvents.java

```diff
@@ -129,27 +85,109 @@ private static boolean isGround(Level world, BlockPos pos) {
         return Main.SERVER_CONFIG.groundTypes.stream().anyMatch(tag -> tag.contains(b.getBlock()));
     }
 
-    private static void destroy(Level world, Player player, BlockPos pos, ItemStack heldItem) {
+    private void destroyTree(Player player, Level world, BlockPos pos) {
+        LinkedList<BlockPos> connectedLogs = scanForTree(world, pos);
+        for (BlockPos log : connectedLogs) {
+            destroy(world, player, log);
+        }
+    }
+
+    private static void destroy(Level world, Player player, BlockPos pos) {
+        ItemStack heldItem = player.getMainHandItem();
         if (heldItem != null) {
             heldItem.getItem().mineBlock(heldItem, world, world.getBlockState(pos), pos, player);
             world.destroyBlock(pos, true);
             player.causeFoodExhaustion(0.025F);
         }
     }
 
-    private static class BlockPosList extends ArrayList<BlockPos> {
-        @Override
-        public boolean add(BlockPos pos) {
-            if (!contains(pos)) {
-                return super.add(pos);
+    public static LinkedList<BlockPos> scanForTree(LevelReader level, BlockPos start) {
+        if (!isLog((Level) level, start)) {
+            return new LinkedList<>();
+        }
+        BlockState state = level.getBlockState(start);
+        boolean[] leavesFound = new boolean[1];
+        LinkedList<BlockPos> result =
+                recursiveSearch(level, start, (pos, bs, isRightBlock) -> {
+                    if (state.is(BlockTags.WARPED_STEMS) && bs.is(BlockTags.WART_BLOCKS)) {
```

Comment:
Detecting leaves wouldn't work for modded blocks that don't inherit from leaves blocks.
Also hardcoding the nether tree variants isn't an elegant solution.